staging/src/k8s.io/apiserver/pkg/cel/library/cost.go

```diff
@@ -258,18 +240,16 @@ func (l *CostEstimator) CallCost(function, overloadId string, args []ref.Val, re
 			unitCost := uint64(1)
 			lhs := args[0]
 			switch lhs.(type) {
-			case cel.Quantity:
-				return &unitCost
-			case cel.IP:
-				return &unitCost
-			case cel.CIDR:
-				return &unitCost
-			case *cel.Format: // Formats have a small max size.
-				return &unitCost
-			case cel.URL: // TODO: Computing the actual cost is expensive, and changing this would be a breaking change
+			case *cel.Quantity, cel.Quantity,
+				*cel.IP, cel.IP,
+				*cel.CIDR, cel.CIDR,
+				*cel.Format,       // Formats have a small max size. Format takes pointer receiver.
```

Comment:
having both pointer and non-pointer on some types, but specifically not other types like format feels like a sharp edge... would it be a problem to put `cel.Format` (non-pointer) in this switch?

Reply:
Fixed